android/src/main/java/io/customer/reactnative/sdk/CustomerIOReactNativeInstance.kt

```diff
@@ -14,4 +24,87 @@ object CustomerIOReactNativeInstance {
     fun setLogLevel(logLevel: CioLogLevel) {
         (logger as ReactNativeConsoleLogger).logLevel = logLevel
     }
+
+    internal fun initializeSDKFromContext(context: Context) {
+        if (CustomerIO.instanceOrNull() != null) return
+
+        try {
+            val preferencesStorage = PreferencesStorage(context = context)
+            initialize(
+                context = context,
+                environment = preferencesStorage.loadEnvironmentSettings(),
+                configuration = preferencesStorage.loadConfigurationSettings(),
+                sdkVersion = preferencesStorage.loadSDKVersion(),
+            )
+            logger.info("Customer.io instance initialized successfully from preferences")
+        } catch (ex: Exception) {
+            logger.error("Failed to initialize Customer.io instance from preferences, ${ex.message}")
+        }
+    }
+
+    @Throws(IllegalArgumentException::class)
+    internal fun initialize(
+        context: Context,
+        environment: Map<String, Any?>,
+        configuration: Map<String, Any?>?,
+        sdkVersion: String?,
+    ): CustomerIO {
+        val siteId = environment.getString(Keys.Environment.SITE_ID)
+        val apiKey = environment.getString(Keys.Environment.API_KEY)
+        val region = environment.getProperty<String>(
+            Keys.Environment.REGION
+        )?.takeIfNotBlank().toRegion()
+        val organizationId = environment.getProperty<String>(
+            Keys.Environment.ORGANIZATION_ID
+        )?.takeIfNotBlank()
+
+        return CustomerIO.Builder(
+            siteId = siteId,
+            apiKey = apiKey,
+            region = region,
+            appContext = context.applicationContext as Application,
+        ).apply {
+            setClient(Client.ReactNative(sdkVersion = sdkVersion ?: "n/a"))
+            setupConfig(configuration)
+            addCustomerIOModule(module = configureModuleMessagingPushFCM(configuration))
+            if (!organizationId.isNullOrBlank()) {
+                addCustomerIOModule(module = configureModuleMessagingInApp(organizationId))
+            }
+        }.build()
+    }
+
+    private fun CustomerIO.Builder.setupConfig(config: Map<String, Any?>?): CustomerIO.Builder {
+        if (config == null) return this
+
+        val logLevel = config.getProperty<Double>(Keys.Config.LOG_LEVEL).toCIOLogLevel()
+        setLogLevel(logLevel = logLevel)
+        setLogLevel(level = logLevel)
```

Comment:
ASAIR, these are two polymorphed functions, one for the react native module and other for the SDK. Please correct me if I am mistaken.

Reply:
Yes, you are correct. But I see the confusion because of same names, so I think we should rename the one in react native module for better readability. Thanks for pointing this out.

Reply:
Agreed. At first it looked like calling same function twice. Thanks for making the changes.